### PR TITLE
fix: removes unused tracking options

### PR DESCRIPTION
**Pull request by kevinpagtakhan**
### Summary

Removes unused tracking options from type definition

Docs update: https://github.com/Amplitude-Developer-Docs/amplitude-dev-center/pull/209

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

### CODE REVIEW

File: packages/analytics-browser/src/config.ts [hunk 1]
```diff
@@ -21,6 +21,15 @@ import { SessionManager } from './session-manager';
 
 export const getDefaultConfig = () => {
   const cookieStorage = new MemoryStorage<UserSession>();
+  const trackingOptions: Required<TrackingOptions> = {
```
Comment (kevinpagtakhan) [inline packages/analytics-browser/src/config.ts:24]: I created this to enforce all tracking options to have a explicit default value

File: packages/analytics-types/src/config.ts [hunk 2]
```diff
@@ -60,23 +60,19 @@ export type InitOptions<T extends Config> =
...
   deviceModel?: boolean;
-  dma?: boolean;
   ipAddress?: boolean;
   language?: boolean;
   osName?: boolean;
   osVersion?: boolean;
   platform?: boolean;
-  region?: boolean;
-  versionName?: boolean;
+  [key: string]: boolean | undefined;
```
Comment (kevinpagtakhan) [inline packages/analytics-types/src/config.ts:70]: I added this to make the changes here backwards compatible for the removed properties

**Review by dnshi — APPROVED** [review]

File: packages/analytics-types/src/config.ts (hunk 2 above)
Comment (qingzhuozhen) [reply packages/analytics-types/src/config.ts:70]: We don't track these in react native?

Comment (kevinpagtakhan) [reply packages/analytics-types/src/config.ts:70]: @qingzhuozhen region and app version being tracked but not controlled by tracking options as of the moment

**Pull request merged**